### PR TITLE
Upgrade to detekt 1.6.0

### LINKED ISSUE #7360: [Meta] Update Detekt version
### Why/User Benefit/User Problem
- The current used version of detekt is an "RC" and not a "GA", plus it's super old (released in Jun 1), we're using 1.0.0-RC16 while the current GA is at 1.3.0

### Acceptance Criteria (Added by PM. For EPM to track when a Meta feature is done)
- Dependency updated. 

### What / Requirements (Added by PM and Eng Manager)
-UX Designs (Immediate task: Assign UX issue to this epic)
-User stories (to be created by PM)
-List dependencies on other issues/teams etc.

PS: should add a "help wanted" and "good first issue" tag on this.

### PR DESCRIPTION
**Pull request by gaul**
Changelog:

https://arturbosch.github.io/detekt/changelog.html

Fixes #7360.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,7 +18,7 @@ buildscript {
 }
 
 plugins {
-    id("io.gitlab.arturbosch.detekt").version("1.0.0-RC16")
+    id("io.gitlab.arturbosch.detekt").version("1.6.0")
```
Comment (dector) [inline build.gradle]: Version in `Dependencies` should be also fixed.

File: build.gradle [hunk 2]
```diff
@@ -80,3 +79,10 @@ task ktlint(type: JavaExec, group: "verification") {
     main = "com.pinterest.ktlint.Main"
     args "app/src/**/*.kt"
 }
+
+task detektProjectBaseline(type: io.gitlab.arturbosch.detekt.DetektCreateBaselineTask) {
```
Comment (dector) [inline build.gradle]: Just a suggestion:
```
tasks.withType(io.gitlab.arturbosch.detekt.Detekt.class) {
}
```

or
```
import io.gitlab.arturbosch.detekt.Detekt

tasks.withType(Detekt.class) {
}
```

File: build.gradle [hunk 3]
```diff
@@ -80,3 +79,10 @@ task ktlint(type: JavaExec, group: "verification") {
...
+    exclude("**/resources/**")
+    exclude("**/build/**")
+    exclude("**/tmp/**")
+    exclude(".*test.*")
```
Comment (dector) [inline build.gradle]: Is this a correct globing pattern?

Comment (gaul) [reply build.gradle]: Changed to match original.

File: build.gradle (hunk 2 above)
Comment (gaul) [reply build.gradle]: Done.

File: build.gradle (hunk 1 above)
Comment (gaul) [reply build.gradle]: Done.

File: build.gradle (hunk 3 above)
Comment (dector) [reply build.gradle]: I meant that `.*test.*` was a regexp, now it'g glob pattern where `.` means `.` but not one symbol. :)

Maybe we should change this to `**/test/**` to exclude test source. And rerun detekt to be sure that it matches everything we need.

Comment (gaul) [reply build.gradle]: Done.

**Comment by ekager** [comment]
Thanks! We are having an issue right now where CI can't run on contributor PRs, so I've opened https://github.com/mozilla-mobile/fenix/pull/9306 with your commit (you'll still get author credit!.

**Pull request closed without merging**